CONTRIBUTING.md

```diff
@@ -0,0 +1,92 @@
+# Contributing
+
+Contributions from the community are essential in keeping Hibernate (and any Open Source
+project really) strong and successful.  
+
+# Legal
+
+You should check the relevant license under which all contributions will be licensed for the specific project you are contributing to.
+
+All contributions are subject to the [Developer Certificate of Origin (DCO)](https://developercertificate.org/).  
+
+The DCO text is available verbatim in the [dco.txt](dco.txt) file.
+
+
+## Guidelines
+
+While we try to keep requirements for contributing to a minimum, there are a few guidelines 
+we ask that you mind.
+
+For code contributions, these guidelines include (**where applicable**):
+* Respect the project code style.
+* Have a corresponding JIRA [issue](https://hibernate.atlassian.net/) and be sure to include the key for this JIRA issue in your commit messages.
```

Comment:
I thought about this: for bigger projects, we have Jira, while for smaller ones, we usually use GitHub issues. But then, a big project will most likely have its own contributing guide, and this common one will most likely show up in smaller projects. Do you think we should somehow point users towards GitHub issues for projects without Jira? 🙈

Reply:
I can mention GH issues alongside Jira; in either case that's why i put that 'where applicable' there for.